app/models/voxelytics/ElasticsearchClient.scala

```diff
@@ -0,0 +1,192 @@
+          "logger_name" -> Json.obj(
+            "type" -> "text",
+            "fields" -> Json.obj("keyword" -> Json.obj("type" -> "keyword", "ignore_above" -> 256))),
+          "run_name" -> Json.obj(
+            "type" -> "text",
+            "fields" -> Json.obj("keyword" -> Json.obj("type" -> "keyword", "ignore_above" -> 256))),
+          "task_name" -> Json.obj(
+            "type" -> "text",
+            "fields" -> Json.obj("keyword" -> Json.obj("type" -> "keyword", "ignore_above" -> 256))),
+          "wk_org" -> Json.obj("type" -> "text",
+                               "fields" -> Json.obj("keyword" -> Json.obj("type" -> "keyword", "ignore_above" -> 256))),
+          "wk_url" -> Json.obj("type" -> "text",
+                               "fields" -> Json.obj("keyword" -> Json.obj("type" -> "keyword", "ignore_above" -> 256))),
+          "workflow_hash" -> Json.obj(
+            "type" -> "text",
+            "fields" -> Json.obj("keyword" -> Json.obj("type" -> "keyword", "ignore_above" -> 256)))
+        ))
+      ))
+  )
+
+  private lazy val serverStartupFuture: Fox[Unit] = {
+    logger.info("Waiting for Elasticsearch to become available.")
+    for {
+      _ <- pollUntilServerStartedUp(LocalDateTime.now.plus(Duration.ofMillis(conf.startupTimeout.toMillis))) ~> 500
+      _ <- bootstrapIndexOnServer
+    } yield ()
+  }
+
+  private def pollUntilServerStartedUp(until: LocalDateTime): Fox[Unit] = {
+    def waitAndRecurse(until: LocalDateTime): Fox[Unit] =
+      for {
+        _ <- akka.pattern.after(POLLING_INTERVAL, using = system.scheduler)(Future.successful(()))
+        _ <- bool2Fox(!LocalDateTime.now().isAfter(until)) ?~> s"Elasticsearch did not become ready within ${conf.startupTimeout}."
+        _ <- pollUntilServerStartedUp(until)
+      } yield ()
+
+    for {
+      isServerAvailableBox <- rpc(s"${conf.uri}/_cluster/health?wait_for_status=yellow&timeout=10s").request
+        .withMethod("GET")
+        .execute()
+        .flatMap(result =>
+          if (Status.isSuccessful(result.status)) {
+            Fox.successful(true)
+          } else {
+            Fox.failure(s"Unexpected error code from Elasticsearch ${result.status}.")
+        })
+        .recoverWith({
+          case e: java.net.ConnectException => {
+            logger.debug(s"Elasticsearch connection exception: $e")
+            Fox.successful(false)
+          }
+          case e =>
+            logger.error(s"Unexpected error $e")
+            Fox.failure("Unexpected error while trying to connect to Elasticsearch.", Full(e))
+        })
+      isServerAvailable <- isServerAvailableBox.toFox
+      _ <- if (!isServerAvailable) {
+        waitAndRecurse(until)
+      } else {
+        logger.info("Elasticsearch is available.")
+        Fox.successful(())
+      }
+    } yield ()
+  }
+
+  private def bootstrapIndexOnServer: Fox[Unit] =
+    for {
+      indexExists <- rpc(s"${conf.uri}/${conf.index}").head.map(_ => true).getOrElse(false)
```

Comment:
Took me a while to read that, I didn’t even know Fox had a getOrElse! But well, fair enough :D